packages/health-check/src/api/index.js

```diff
@@ -2,61 +2,75 @@ const { StatusCodes } = require("http-status-codes");
 const { sum, sumBy } = require("lodash");
 const db = require("../db");
 
-// getStatus returns the server's current health check status
-function getStatus() {
-  const disabled = db.get("disabled").value();
+/**
+ * Get status code that should be returned in the API response.
+ * - OK (200) in case everything is healthy
+ * - SERVICE_UNAVAILABLE (503) in case of any failures or if disabled
+ */
+function getStatusCode() {
+  // check whether the portal has been manually disabled
+  const disabled = getDisabled();
 
   if (disabled) {
     return StatusCodes.SERVICE_UNAVAILABLE;
   }
 
-  // Grab entry element from DB
-  const entry = db.get("entries").orderBy("date", "desc").head().value();
+  // grab one most recent critical entry element from DB
+  const entry = getCurrentCriticalEntry();
 
-  // Check that every critical check entry is up
-  if (entry && entry.checks.every(({ up, critical }) => up && critical)) {
+  // find out whether every check in the entry is up
+  if (entry && entry.checks.every(({ up }) => up)) {
     return StatusCodes.OK;
   }
 
-  // At least one check failed
+  // in case at least one check failed
   return StatusCodes.SERVICE_UNAVAILABLE;
 }
 
-// getTimeout returns the average time out from a sample of 10 health check
-// entries.
-function getTimeout() {
-  if (getStatus() === StatusCodes.SERVICE_UNAVAILABLE) {
-    return 0;
-  }
-
-  // Grab 10 entries from the database as a sample to determine the average
-  // timeout for the server.
+/**
+ * Get the sample of most recent critical entries and
+ * calculate the avarage response time of all of them
+ */
+function getAvarageResponseTime() {
+  // get most recent 10 successfull checks for the calculation
   const sample = db
-    .get("entries")
+    .get("critical")
```

Comment:
not really sure I'm a big fan of the split between `critical` entries and `verbose` ones, why didn't we just `filter` on `critical: true`

Reply:
The PR description should help with that.